storage-bigtable/src/lib.rs

```diff
@@ -377,26 +381,56 @@ pub enum CredentialType {
 }
 
 #[derive(Debug)]
-pub struct LedgerStorageConfig {
+pub struct LedgerStorageConfig<C = HttpConnector> {
     pub read_only: bool,
     pub timeout: Option<std::time::Duration>,
     pub credential_type: CredentialType,
     pub instance_name: String,
     pub app_profile_id: String,
+    pub connector: Option<C>,
 }
 
-impl Default for LedgerStorageConfig {
+impl Default for LedgerStorageConfig<HttpConnector> {
     fn default() -> Self {
         Self {
             read_only: true,
             timeout: None,
             credential_type: CredentialType::Filepath(None),
             instance_name: DEFAULT_INSTANCE_NAME.to_string(),
             app_profile_id: DEFAULT_APP_PROFILE_ID.to_string(),
+            connector: None,
         }
     }
 }
 
+impl LedgerStorageConfig<hyper_proxy::ProxyConnector<HttpConnector>> {
+    pub fn with_proxy_connector(&mut self, proxy_uri: Option<String>) -> Result<()> {
+        let proxy_uri = match (proxy_uri, std::env::var("BIGTABLE_PROXY")) {
+            (Some(proxy_uri), _) => proxy_uri,
+            (None, Ok(proxy_uri)) => proxy_uri,
+            (None, Err(err)) => return Err(Error::InvalidUri("".to_string(), err.to_string())),
+        };
+
+        let proxy = hyper_proxy::Proxy::new(
+            hyper_proxy::Intercept::All,
+            proxy_uri
+                .parse::<http::Uri>()
+                .map_err(|err| Error::InvalidUri(proxy_uri, err.to_string()))?,
+        );
+
+        let mut http = HttpConnector::new();
+        http.enforce_http(false);
+        http.set_nodelay(true);
```

Comment:
Any particular reason for this addition?

Reply:
I actually described this in my other PR: https://github.com/solana-labs/solana/pull/26398
In `tonic` which is used here in the inner connector `nodelay` is set to true.

Reply:
Ah, missed that one. Thanks for the link